misc/python/materialize/git.py

```diff
@@ -15,7 +15,12 @@
 from pathlib import Path
 from typing import List, Optional, Set, Union
 
-import semver.version
+import semver
+
+if semver.__version__.startswith("2."):
```

Comment:
Is there a reason we're potentially using different versions of the `semver` package here, rather than just pinning the one we prefer in our requirements file?

Reply:
Yes, unfortunately there is. I would love to use the same version but the materialize repo already uses version 3.0+ while the cloud repo requires a 2.x version due to its dependencies (`pulumi` and `launchdarkly-server-sdk`). More details can be found in the [questions section of the design doc](https://github.com/MaterializeInc/materialize/blob/6e733ac90fef487e2de6dc31159f163a4d9a14f7/doc/developer/design/20230613_cloudtest_improvements.md#how-to-handle-semver-dependency-conflict).

Reply:
I will merge it now as it is and revisit this after my holidays (in order to avoid inadvertently breaking anything non-obvious by downgrading)

Reply:
I had a look at it and discussed it with @def-. While it might be possible to do the downgrade, it has the disadvantage that the older semver version provides no type information. Therefore, we prefer to stick to this version and keep the workaround.